### PR TITLE
doc: Update 54H architecture reset doc

### PR DESCRIPTION
**Pull request by FrancescoSer**
Updated 54H architecture reset doc.

### CODE REVIEW

**Comment by FrancescoSer** [comment]
@hubertmis  @bjorgan Could you check if this edit now answers your concerns?

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_architecture_reset.rst [hunk 1]
```diff
@@ -60,16 +64,21 @@ The following is the reset behavior when the LCS of the nRF54H20 SoC is either i
...
        The register used is ``RESET`` (address offset: ``0x00``).
    * - ``RESET_PIN``
      - J-Link toggles the ``RESET`` pin via OBD.
+   * - ``RESET_VIA_SECDOM``
+     - In the Secure Domain, this reset is mapped to the ``SYS_RESET_REQUEST`` bit, as described in the ARM specification for the *Application Interrupt and Register Controller*.
+       This reset sends a reset request over ADAC to the Secure Domain Firwmare (SDFW).
+       This triggers a reset of the Secure Domain, which subsequently resets the entire system.
+
+       In the local domains, this reset is available only if the other domains have a valid configuration in the ``CPUCONF`` register and their access port is unprotected.
+       The PC tool uses the CTRL-AP mailbox to send a local domain reset request to the Secure Domain Firmware (SDFW).
```
Comment (bjorgan) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_architecture_reset.rst:79]: I wasn't precise enough in https://github.com/nrfconnect/sdk-nrf/pull/17411#discussion_r1802937633, sorry about that. There is still some incorrect info here, but I think we can just reduce it down to this:

```suggestion
       - Uses the CTRL-AP mailbox to send a local domain reset request to the Secure Domain Firmware (SDFW).
```

If a more detailed description about what SDFW actually does is needed, https://github.com/nrfconnect/sdk-secdom/blob/main/doc/ctrlap_adac.rst#reset---sdfw_adac_reset has a description of it. Copypasting from that, something like 

"  - Uses the CTRL-AP mailbox to send a local domain reset request to the Secure Domain Firmware (SDFW).

SDFW performs a reset of nonessential local domains and global peripherals owned by those domains. Nonessential domains include all local domains except for the Secure Domain. Only domains with a populated UICR are reset. The reset CPUs do not restart after the command is completed."

might work here.

Comment (FrancescoSer) [reply doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_architecture_reset.rst]: Solved, Please let me know if now the description is correct @bjorgan

**Review by bjorgan — APPROVED** [review]

**Comment by ingeryp** [comment]
I think this generally looks great :)

Just out of curiosity:
I know there is some risk of confusion regarding the LCS terminology, as we define separate life cycle states for different domains:
* Secure Domain / Chip - Enforced by ROM code. The LCS that is being talked about in the `life cycle state` and `reset behavior` chapters
* Application Domain - Enforced by Secure Domain FW.
* Radio Domain - Enforced by Secure Domain FW.

How do / will you disambiguate these in documentation?

**Review by ingeryp — APPROVED** [review]

**Pull request merged**